packages/venia-ui/lib/components/CategorySort/categorySort.js

```diff
@@ -0,0 +1,125 @@
+import React, { useMemo, useState } from 'react';
+import { Check } from 'react-feather';
+import uuidV4 from 'uuid/v4';
+import { arrayOf, shape, string, func, oneOf } from 'prop-types';
+
+import defaultClasses from './categorySort.css';
+import { mergeClasses } from '../../classify';
+import Icon from '../Icon/icon';
+
+const CategorySort = props => {
+    const classes = mergeClasses(defaultClasses);
+    const { availableSortMethods, sortControl } = props;
+    const { currentSort, setSort } = sortControl;
+
+    const [visible, setVisible] = useState(false);
+    const [sortAttribute, setSortAttribute] = useState(
+        currentSort.sortAttribute
+    );
+    const [sortDirection, setSortDirection] = useState(
+        currentSort.sortDirection
+    );
+
+    const sortElements = useMemo(() => {
+        const onChildButtonClick = sortAttribute => {
+            setSort({
+                sortAttribute: sortAttribute.attribute,
+                sortDirection: sortAttribute.sortDirection
+            });
+            setVisible(false);
+            setSortAttribute(sortAttribute.attribute);
+            setSortDirection(sortAttribute.sortDirection);
+        };
+
+        if (visible === false) {
+            return;
+        }
+
+        const sortElements = [];
+        for (const element of availableSortMethods) {
+            const key = uuidV4();
```

Comment:
This may circumvent the error, but it defeats the purpose of keys, especially if not memoized. As I suggested in an earlier comment, you should derive a string key from each of these objects, which can be used for the `key` prop as well as for storing the current sort value in state.

I suggest concatenating `attribute` and `sortDirection`, which will give you a unique string that's representative of the object. Since you'll want this key for multiple purposes, it may serve you better to transform the array into a map right away, as we often do in this codebase.

```es6
const availableSortMethods = new Map()

for (method of props.availableSortMethods) {
  // derive a unique key for each method
  const key = `${attribute}--${sortDirection}`

  // add each method to the map
  availableSortMethods.set(key, method)
}
```